[Regression@293947@main] [WebGPU] Some external sites report missing geometry

mwyrzykowski opened the pull request:
#### 7c88a46d753b72ac9f2c0c1767b7e26d9566c0c2
<pre>
[Regression@293947@main] [WebGPU] Some external sites report missing geometry
<a href="https://bugs.webkit.org/show_bug.cgi?id=294360">https://bugs.webkit.org/show_bug.cgi?id=294360</a>
<a href="https://rdar.apple.com/153138317">rdar://153138317</a>

Reviewed by Tadeu Zagallo.

We were failing to update a cache resulting in
calls to setVertexBuffer: getting skipped.

But this cache was not really useful, since setVertexBuffer() already
maintains a cache itself. So better to remove the almost duplicate cache.

* Source/WebGPU/WebGPU/RenderPassEncoder.h:
* Source/WebGPU/WebGPU/RenderPassEncoder.mm:
(WebGPU::RenderPassEncoder::setVertexBuffer):
(WebGPU::RenderPassEncoder::setVertexBytes):
(WebGPU::RenderPassEncoder::errorValidatingAndBindingBuffers):
(WebGPU::RenderPassEncoder::executeBundles):
(WebGPU::RenderPassEncoder::setPipeline):

* LayoutTests/fast/webgpu/regression/repro_294360-expected.txt: Added.
* LayoutTests/fast/webgpu/regression/repro_294360.html: Added.
Add regression test.

Canonical link: <a href="https://commits.webkit.org/296192@main">https://commits.webkit.org/296192@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/fef25fbb771ba8c9e17346f641a408cc622b3bb0

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/107605 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/27288 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/17699 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/112822 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/58147 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/27980 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/35790 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/81701 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/110534 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/22174 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/96986 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/62081 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/21610 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/15125 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/57586 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/91545 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/15159 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/115923 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/34673 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/25570 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/90733 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/35049 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/93236 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/90474 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23081 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/35391 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/13175 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/30436 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/34594 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/40150 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/34340 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/37700 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/36003 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/8807f61b507a270a6a10588e6e7a479b6f18c71b. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/46646

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/d77094f9ec0c7cb91a0de3a7f5a3fc5910ecab5c. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/46646

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/fef25fbb771ba8c9e17346f641a408cc622b3bb0. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/46646

mwyrzykowski committed:
[Regression@293947@main] [WebGPU] Some external sites report missing geometry
https://bugs.webkit.org/show_bug.cgi?id=294360
rdar://153138317

Reviewed by Tadeu Zagallo.

We were failing to update a cache resulting in
calls to setVertexBuffer: getting skipped.

But this cache was not really useful, since setVertexBuffer() already
maintains a cache itself. So better to remove the almost duplicate cache.

* Source/WebGPU/WebGPU/RenderPassEncoder.h:
* Source/WebGPU/WebGPU/RenderPassEncoder.mm:
(WebGPU::RenderPassEncoder::setVertexBuffer):
(WebGPU::RenderPassEncoder::setVertexBytes):
(WebGPU::RenderPassEncoder::errorValidatingAndBindingBuffers):
(WebGPU::RenderPassEncoder::executeBundles):
(WebGPU::RenderPassEncoder::setPipeline):

* LayoutTests/fast/webgpu/regression/repro_294360-expected.txt: Added.
* LayoutTests/fast/webgpu/regression/repro_294360.html: Added.
Add regression test.

Canonical link: https://commits.webkit.org/296192@main

webkit-commit-queue commented:
Committed 296192@main (7c88a46d753b): <https://commits.webkit.org/296192@main>

Reviewed commits have been landed. Closing PR #46646 and removing active labels.
